Make Pacman great again

mvazquezc opened the pull request:
This PR addresses some things:
- Updates jquery to the latest version
- Now Pacman works on iOS devices (it seems window.navigator.vibrate(200); it's not supported on iOS devices)
- Cloud detection now works faster when running on k8s (which is where we usually demo the app)

mvazquezc committed:
Updated jquery version + deleted incompatible iOS browser event so controls work on iOS

mvazquezc committed:
Added Ryan's cloud detection improvement

mvazquezc committed:
Fixed typo

mvazquezc commented:
@cooktheryan

font commented on routes/location.js:
```diff
@@ -28,8 +28,8 @@ router.get('/metadata', function(req, res, next) {
-    // Try Openstack first
-    getOpenStackCloudMetadata(function(err, c, z) {
+    // Query k8s node api
+    getK8sCloudMetadata(function(err, c, z) {
```
I really need to parallelize these function calls.

font commented on public/pacman-canvas.js:
```diff
@@ -1469,22 +1469,18 @@ function checkAppCache() {
         // Mobile Control Buttons
         $(document).on('touchend mousedown','#up',function(event) {
             event.preventDefault();
-            window.navigator.vibrate(200);
```
Does this effectively remove the vibration feedback when using the navigation buttons on Android devices? I think the vibration is cool feedback when using the buttons to control pacman and enhances the gaming experience. Do you know if there's an equivalent vibration call to enable this for iOS devices? If not, could we check if we're running on an Android device to invoke the vibrate function call?

mvazquezc replied:
You're correct, it removes the vibration feedback when using the navigation buttons. 

I just sent a new commit including your suggestion, now it detects if the browser supports vibration before calling the vibrate function.

It seems that iOS doesn't support this feature: 
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/vibrate

Now it will vibrate on Android when supported by the browser.

font replied:
Awesome, nice work!

font reviewed (commented):
Thank you @mvazquezc for this update! Nice to see iOS finally being fixed. :smile: I just had a few questions about the use of the vibration feedback.

mvazquezc committed:
Added vibration back

mvazquezc commented:
@font This last commit includes:
- Moved the vibration to a function that will fire the event only if the browser supports vibration
- Fixed a bug when querying the k8s api on OCP3.X, the cloudname and az were returned empty, now we will error if that's the case

The parallelization of getCloudMetadata is a nice to have, but atm it works pretty fast anyway, up to you if you want to include that.

font commented on routes/location.js:
```diff
@@ -395,23 +396,27 @@ function getK8sCloudMetadata(callback) {
-            	cloudName = String(provider.split(":", 1)); // Split on providerID if request was successful
-	    }
-
+	        if (metaData.spec.providerID) {
```
nit: is this a spacing change only? I'm not sure it lines up with the other logic.

mvazquezc replied:
It seems so, it was showing okay in my editor, it's fixed now.

mvazquezc committed:
Fixed spacing

font reviewed (commented):
LGTM. Thanks @mvazquezc!

font commented:
@mvazquezc I've created and pushed a new container image that includes these changes `quay.io/ifont/pacman-nodejs-app:latest`.
